packages/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -12,17 +13,42 @@ import type { DialogProps, DialogState } from './Dialog.types';
  * @param ref - reference to root HTMLElement of Dialog
  */
 export const useDialog_unstable = (props: DialogProps, ref: React.Ref<HTMLElement>): DialogState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { overlay, open = false, type = 'modal' } = props;
+  const isNonModal = type === 'non-modal';
+
+  const contentRef = React.useRef(null);
+  const { modalAttributes } = useModalAttributes({ trapFocus: !isNonModal && true });
+  const { findFirstFocusable } = useFocusFinders();
+
+  React.useEffect(() => {
+    if (open && contentRef.current) {
+      const firstFocusable = findFirstFocusable(contentRef.current);
+      firstFocusable?.focus();
+    }
+  }, [open, contentRef, findFirstFocusable]);
+
+  const state: DialogState = {
+    open,
+    type,
     components: {
-      // TODO add slot types here if needed (div is the default)
       root: 'div',
+      overlay: 'div',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
+
     root: getNativeElementProps('div', {
-      ref,
+      ref: useMergedRefs(ref, contentRef),
+      role: 'dialog',
+      'aria-modal': !isNonModal && true,
+      'aria-label': props['aria-labelledby'],
+      ...modalAttributes,
       ...props,
     }),
+    overlay: resolveShorthand(overlay),
   };
+
+  if (type === 'alert') {
+    state.root.role = 'alertdialog';
+  }
```

Comment:
nit: I would prefer to hoist this condition direction to when you declare the props
```suggestion
```

Reply:
Suggestion here
https://github.com/microsoft/fluentui/pull/22051/files#r824521503